core/src/main/java/org/stellar/anchor/util/SepLanguageHelper.java

```diff
@@ -0,0 +1,105 @@
+package org.stellar.anchor.util;
+
+import static org.stellar.anchor.util.Log.errorEx;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import org.stellar.anchor.api.exception.SepValidationException;
+import org.stellar.anchor.config.AppConfig;
+
+public class SepLanguageHelper {
+  static Map<String, Iso3316Language> fallbackLangs = null;
+
+  public static String validateLanguage(AppConfig appConfig, String lang) {
+    if (fallbackLangs == null) {
+      fallbackLangs = prepareFallbackLanguages(appConfig);
+    }
+
+    if (lang != null) {
+      List<String> languages = appConfig.getLanguages();
+      if (languages != null && languages.size() > 0) {
+        if (languages.stream().noneMatch(l -> l.equalsIgnoreCase(lang))) {
+          return getFallbackLanguage(lang);
+        }
+      }
+      return lang;
+    } else {
+      return getFallbackLanguage(lang);
+    }
+  }
+
+  public static void reset() {
+    fallbackLangs = null;
+  }
+
+  static Map<String, Iso3316Language> prepareFallbackLanguages(AppConfig appConfig) {
+    Map<String, Iso3316Language> fallbacks = new HashMap<>();
+    appConfig
+        .getLanguages()
+        .forEach(
+            lang -> {
+              try {
+                Iso3316Language isoLang = Iso3316Language.of(lang);
+                fallbacks.putIfAbsent(isoLang.getLangKey(), isoLang);
+              } catch (SepValidationException ex) {
+                errorEx(ex);
+              }
+            });
+    return fallbacks;
+  }
+
+  static final String LANGUAGE_ONLY_DEFAULT = "en";
+  static final String LANGUAGE_DEFAULT = "en-US";
```

Comment:
💡 The word `only` is a bit ambiguous to me. What about renaming these two to:
LANGUAGE_WITHOUT_LOCALE_DEFAULT
LANGUAGE_WITH_LOCALE_DEFAULT

Reply:
Changed.